File: apps/api/src/domain/identity/value-objects/UserStatus.ts
export enum UserStatusValue {
  ACTIVE = 'active',
  BLOCKED = 'blocked',
  DELETED = 'deleted',
}

export class UserStatus {
  private constructor(public readonly value: UserStatusValue) {}

  static readonly ACTIVE = new UserStatus(UserStatusValue.ACTIVE);
  static readonly BLOCKED = new UserStatus(UserStatusValue.BLOCKED);
  static readonly DELETED = new UserStatus(UserStatusValue.DELETED);

  static fromValue(value: string): UserStatus {
    switch (value) {
      case UserStatusValue.ACTIVE:
        return UserStatus.ACTIVE;
      case UserStatusValue.BLOCKED:
        return UserStatus.BLOCKED;
      case UserStatusValue.DELETED:
        return UserStatus.DELETED;
      default:
        throw new Error(`Unknown user status: ${value}`);
    }
  }

  isActive(): boolean {
    return this.value === UserStatusValue.ACTIVE;
  }

  equals(other: UserStatus): boolean {
    return this.value === other.value;
  }
}

File: backend/domain/identity/value_objects/user_status.py
"""
UserStatus Value Object.
"""
from domain.shared.value_object import ValueObject


class UserStatus(ValueObject):
    """Value Object для статуса пользователя."""
    
    def __init__(self, value: str):
        if value not in ['active', 'blocked', 'deleted']:
            raise ValueError(f"Invalid user status: {value}")
        self._value = value
    
    def is_active(self) -> bool:
        return self._value == 'active'
    
    @property
    def value(self) -> str:
        return self._value
    
    # Предопределенные статусы
    ACTIVE = None
    BLOCKED = None
    DELETED = None


UserStatus.ACTIVE = UserStatus('active')
UserStatus.BLOCKED = UserStatus('blocked')
UserStatus.DELETED = UserStatus('deleted')
